Client/Frontend/Browser/HomePanel/NTPDownloader.swift

```diff
@@ -0,0 +1,215 @@
+                guard let self = self else { return }
+                
+                if let error = error {
+                    return completion(nil, error)
+                }
+                
+                guard let data = data else {
+                    return completion(nil, "Invalid Photos.json for NTP Download")
+                }
+                
+                self.unzip(data: data) { url, error in
+                    completion(url, error)
+                }
+            }
+        } else {
+            self.download(path: "photo.json") {[weak self] data, error in
+                guard let self = self else { return }
+                
+                if let error = error {
+                    return completion(nil, error)
+                }
+                
+                guard let data = data else {
+                    return completion(nil, "Invalid Photos.json for NTP Download")
+                }
+                
+                do {
+                    let item = try JSONDecoder().decode(NTPItemInfo.self, from: data)
+                    self.unzip(item: item) { url, error in
+                        completion(url, error)
+                    }
+                } catch {
+                    completion(nil, error)
+                }
+            }
+        }
+    }
+    
+    private func getBaseURL() -> URL? {
+        guard let url = URL(string: NTPDownloader.BASE_URL) else {
+            return nil
+        }
+        
+        if self.isZipped {
+            return url.appendingPathComponent(getSupportedLocale())
+                      .appendingPathExtension("zip")
+        }
+        
+        return url.appendingPathComponent(getSupportedLocale())
+    }
+    
+    private func getSupportedLocale() -> String {
+        guard let region = Locale.current.regionCode else {
+            return self.defaultLocale
+        }
+        
+        if supportedLocales.contains(region) {
+            return region
+        }
+        
+        return self.defaultLocale
+    }
+    
+    //MARK: - Download & Unzipping
+    
+    // Downloads the item at the specified url relative to the BASE_URL
+    private func download(path: String?, _ completion: @escaping (Data?, Error?) -> Void) {
+        guard var url = self.getBaseURL() else {
+            return completion(nil, nil)
+        }
+        
+        if let path = path {
+            url = url.appendingPathComponent(path)
+        }
+        
+        let request = URLRequest(url: url)
+        URLSession(configuration: .ephemeral).dataRequest(with: request) { data, response, error in
+            if let error = error {
+                return completion(nil, error)
+            }
+            
+            guard let response = response as? HTTPURLResponse else {
+                return completion(nil, error)
+            }
+            
+            if response.statusCode < 200 || response.statusCode > 299 {
+                return completion(nil, nil)
```

Comment:
Would be good to return some error here containing status code, can return a simple String, as String conforms to error in our code(see `ErrorExtensions.swift`)